E-commerce tutorial

shannonbux opened the pull request.

shannonbux commented on docs/docs/ecommerce-tutorial/index.md:
````diff
@@ -0,0 +1,542 @@
+Create a new Gatsby project by running the `gatsby new` command in the terminal and change directories into the new project you just started:
+
+```shell
+gatsby new ecommerce-gatsby-tutorial https://github.com/gatsbyjs/gatsby-starter-default
````
Is the URL necessary? I thought gatsby new already used the default starter w/o needing a URL

m-allanson replied:
You're right the URL can be removed.

shannonbux commented on docs/docs/ecommerce-tutorial/index.md:
```diff
@@ -0,0 +1,542 @@
+
+There is a default checkout modal that is available through the plugin we’re using, shown in the image below. When we call the `.open()` method through Stripe Checkout, the rest of the screen is darkened and a modal appears over the top, directing the user’s attention to the checkout form. You’ll need to create a button component that calls this method or triggers Stripe from your site.
+
+> **NOTE**: If you’d like to customize the checkout modal, see the [Stripe Checkout Elements plugin](/packages/gatsby-plugin-stripe-elements/?=stripe)
```
Do we need to mention the Stripe Source Plugin here? I don't understand what that plugin does.

https://www.gatsbyjs.org/packages/gatsby-source-stripe/?=stripe

shannonbux commented on docs/docs/ecommerce-tutorial/index.md:
```diff
@@ -0,0 +1,542 @@
+
+> Notice the input fields for email and credit card information—this design is already made by Stripe for you to use without styling anything. 
+
+In order to launch the Stripe Checkout modal (shown above), you need to call the `.configure()` method from Stripe Checkout in your Gatsby site. The only information you are required to provide is your publishable API key so that Stripe knows what account on their platform to send payments to. You can read about other recommended and optional configurations like currency or shipping information i ...
```
I'm confused; the paragraph above says we need to call `.open()` method and this one says `.configure()`. Which one is it?

m-allanson replied:
Hmm.. It looks like both. `configure()` sets up Stripe and I think it automatically runs every time the page loads. We run `open()` when the buy button is clicked, which then triggers the Stripe payment overlay to open.

Maybe we could add a section before the _Creating a button_ heading that describes this sequence of events? 

- _page load_ -> `configure()`
- _some time later..._
- _user clicks button_ -> `open()`

shannonbux commented on docs/docs/ecommerce-tutorial/index.md:
```diff
@@ -0,0 +1,542 @@
+Open gatsby-stripe-serverless-backend in your code editor.
+
+Rename the `secrets.example.json` file to `secrets.json`.
+Replace the string that says "sk_test_STRIPE_SECRET_KEY" in `secrets.json` with your secret key from your Stripe account, and keep the quotation marks around it
```
does it matter if this is test vs production version?

m-allanson replied:
The test version will allow orders to go through without needing a real credit card.

m-allanson commented on docs/docs/ecommerce-tutorial/index.md:
```diff
@@ -0,0 +1,551 @@
+
+to the end of the `<body>` tag across all of your pages in your Gatsby project, allowing you to call the Stripe checkout methods from Stripe’s API to create charges.
+
+If you’d like more extensive access to Stripe data, explore more Gatsby / Stripe plugins.
```
What about something like this?

_If you want to further customise the checkout process or pull Stripe data into your site, check out [Gatsby's plugin library for more Stripe plugins](https://www.gatsbyjs.org/plugins/?=stripe) ._

And then delete the note below about the Stripe Checkout Elements plugin.

m-allanson commented on docs/docs/ecommerce-tutorial/index.md:
```diff
@@ -0,0 +1,551 @@
+
+The openStripeCheckout() function gives additional information to Stripe as a user launches the checkout modal, and will send the information they input to our serverless function once we create it.
+
+The tags in the render() function define the structure of HTML elements that lay out how the component is structured.
```
These paragraphs are missing backticks around the functions `likeThis()`

m-allanson commented on docs/docs/ecommerce-tutorial/index.md:
````diff
@@ -0,0 +1,551 @@
+
+export default IndexPage
+
+```
````
Maybe this code example and the previous could be combined into one?

m-allanson commented on docs/docs/ecommerce-tutorial/index.md:
````diff
@@ -0,0 +1,551 @@
+
+```
+
+If you go back to localhost:8000 in your browser and you have `gatsby develop` running, you should have a big, enticing button on a card where the filler text used to be.
````
Turn localhost:8000 into a clickable link?

m-allanson commented on docs/docs/ecommerce-tutorial/index.md:
```diff
@@ -0,0 +1,542 @@
+
+You can read more about the configurations you’ll use in the `serverless.yml` file in the repo you just cloned or in the Serverless docs. 
+
+`Serverless.yml`
```
👍

m-allanson commented on docs/docs/ecommerce-tutorial/index.md:
```diff
@@ -0,0 +1,551 @@
+
+* Create or login to your Amazon Web Services Account and go to the Identity & Access Management (IAM) page.
+* Click on Users and then Add user. Enter a name in the first field to remind you this User is the Framework, like serverless-admin. Enable Programmatic access by clicking the checkbox. Click Next to go through to the Permissions page. Click on Attach existing policies directly. Search for and select AdministratorAccess then click Next: Review. Check everything looks good and click Create user. L ...
+* View and copy the API Key & Secret to a temporary place and place them into the command at the beginning of this section of the tutorial before running the command.
```
This says _these steps taken from the Serverless AWS Credentials docs_. Maybe this section could be replaced with a link to the relevant AWS docs?

shannonbux commented:
@m-allanson addressed all your comments. Thank you so much for the thoughtful responses to my questions, and let me know if you think the table of contents and the few things I did to shorten it look ok!

shannonbux commented:
@calcsam I think this is ready to merge. Want to take a look or shall I just push the button??

calcsam commented on docs/docs/ecommerce-tutorial/index.md:
```diff
@@ -0,0 +1,524 @@
+
+* Stripe account: [register for an account here](https://dashboard.stripe.com/register)
+* AWS account (free tier that covers anywhere from several thousand to a million requests per month): [register for an account here](https://aws.amazon.com/free/?sc_channel=PS&sc_campaign=acquisition_US&sc_publisher=google&sc_medium=cloud_computing_b&sc_content=aws_account_e_control_q32016&sc_detail=create%20an%20aws%20account&sc_category=cloud_computing&sc_segment=102882721242&sc_matchtype=e&sc_ ...
+
```
I'd add "Willingness to navigate around janky AWS UIs" here just to warn people that there be a few dragons. This feels higher level of difficulty than the rest of our tutorials -- do we want to say that this is a tutorial for developers that have a bit of experience and it may be challenging for newcomers?

shannonbux replied:
Sure, both great ideas. I'll do that right now then merge this!

calcsam commented:
This looks great and I'm excited to merge this in! Hopefully when we get per-page feedback working we will be able to see if folks are getting stuck anywhere.
